### PR TITLE
Balder/control environment var MERGEOK

### PR DESCRIPTION
**Pull request by baldersheim**
@bratseth and @hmusum and @gjoranv PR

### CODE REVIEW

**Review by bratseth — APPROVED** [review]
I agree it would be better to use XML syntax for this, other than that LGTM

**Comment by jasonchaw** [comment]
Hi folks,

Thank you! :-)

**Review by hmusum — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/VespaDomBuilder.java
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/VespaDomBuilder.java]: - Rename to e.g. `ENVIRONMENT_VARIABLES_ELEMENT`
- Only used in ContainerModelBuilder, so move it there

File: config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java
Comment (gjoranv) [inline config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java]: 'search' is redundant, so remove it.

File: config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java
Comment (gjoranv) [inline config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java]: 'jdisc' is soon to be deprecated, so use 'container' instead. (I know a lot of tests still use 'jdisc', but it will be fixed.) Verify that the test still works - you probably need to update the configId in the `getConfig` call below.

**Review by gjoranv — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments.

File: config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java
Comment (baldersheim) [reply config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java]: Fixed

File: config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java
Comment (baldersheim) [reply config-model/src/test/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilderTest.java]: Fixed

**Review by gjoranv — APPROVED** [review]
👍

**Comment by jasonchaw** [comment]
Hi!

Good day!

Can we merge this PR when convenient?

Thanks! :-)

**Comment by baldersheim** [comment]
It will be merged once the merge stop related to non vespa 7 related commits are lifted. ETA is today or tomrrow.

**Comment by jasonchaw** [comment]
Thank you! :-)

**Review by bratseth — APPROVED** [review]

**Pull request merged**
